### PR TITLE
Xamarin.android TodoMvc

### PR DESCRIPTION
**Pull request by vinodsantharam**
Hi Guillaume,

I have developed a Xamarin Android application using redux.NET. 
I have moved state classes into a PCL project.
I think that we can refactor some business logic codes. 
I didn't refactor the Universal App project to include as reference the PCL project.

Any comments will be appreciate :)
Thanks.

Vinod

### CODE REVIEW

File: examples/todomvc/Redux.TodoMvc/Redux.TodoMvc.csproj [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    <Compile Include="States\Todo.cs" />
+    <Compile Include="States\TodosFilter.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Newtonsoft.Json, Version=7.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\packages\Newtonsoft.Json.7.0.1\lib\portable-net40+sl5+wp80+win8+wpa81\Newtonsoft.Json.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Redux">
+      <HintPath>C:\Users\vinod.santharam\.nuget\packages\Redux.NET\0.1.0\lib\portable-net45+netcore45\Redux.dll</HintPath>
```
Comment (GuillaumeSalles) [inline examples/todomvc/Redux.TodoMvc/Redux.TodoMvc.csproj:50]: Should be referenced via nuget.

File: examples/todomvc/Redux.TodoMvc.Android/Redux.TodoMvc.Android.csproj [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <AndroidUseSharedRuntime>False</AndroidUseSharedRuntime>
+    <AndroidLinkMode>SdkOnly</AndroidLinkMode>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Mono.Android" />
+    <Reference Include="mscorlib" />
+    <Reference Include="Redux">
+      <HintPath>C:\Users\vinod.santharam\.nuget\packages\Redux.NET\0.1.0\lib\portable-net45+netcore45\Redux.dll</HintPath>
```
Comment (GuillaumeSalles) [inline examples/todomvc/Redux.TodoMvc.Android/Redux.TodoMvc.Android.csproj:47]: Should be referenced via nuget.

File: examples/todomvc/Redux.TodoMvc.Android/Redux.TodoMvc.Android.csproj [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    </Reference>
+    <Reference Include="System" />
+    <Reference Include="System.Collections.Immutable">
+      <HintPath>C:\Users\vinod.santharam\.nuget\packages\System.Collections.Immutable\1.1.37\lib\dotnet\System.Collections.Immutable.dll</HintPath>
```
Comment (GuillaumeSalles) [inline examples/todomvc/Redux.TodoMvc.Android/Redux.TodoMvc.Android.csproj:51]: Should be referenced via nuget.

**Comment by GuillaumeSalles** [comment]
Great job Vinod ! Using a PCL project is the best move to share code across platforms ! 

I'm going to update the Win10 app to use your PCL project.

The android project does not build on my side. It seems that you added references to Redux and System.Collections.Immutable with an absolute path. ( Resharper ? ;) ) 
I will fix that myself after the merge back to master.

Thanks a lot for your contribution !

**Comment by vinodsantharam** [comment]
:+1: 
It seems that I have some issues with my nuget.config. Sorry for the trouble. I will update my branch :)

**Comment by GuillaumeSalles** [comment]
No problem. It's already done and merge back. ;) 

I just updated the Win10 app to work with the PCL project. Works like a charm ! Thanks again !

https://github.com/GuillaumeSalles/redux.NET/pull/6

**Comment by vinodsantharam** [comment]
Awesome !

**Pull request merged**